src/sys/windows/afd.rs

```diff
@@ -0,0 +1,237 @@
+                &mut iosb,
+                null_mut(),
+                0 as ULONG,
+                FILE_SHARE_READ | FILE_SHARE_WRITE,
+                FILE_OPEN,
+                0 as ULONG,
+                null_mut(),
+                0 as ULONG,
+            );
+            if status != STATUS_SUCCESS {
+                return Err(io::Error::from_raw_os_error(
+                    RtlNtStatusToDosError(status) as i32
+                ));
+            }
+            let fd = File::from_raw_handle(afd_helper_handle as RawHandle);
+            let afd = Afd { fd };
+            let token = NEXT_TOKEN.fetch_add(1, Ordering::Relaxed) + 1;
+            cp.add_handle(token, &afd.fd)?;
+            match SetFileCompletionNotificationModes(
+                afd_helper_handle,
+                FILE_SKIP_SET_EVENT_ON_HANDLE,
+            ) {
+                0 => Err(io::Error::last_os_error()),
+                _ => Ok(afd),
+            }
+        }
+    }
+
+    pub fn poll(
+        &self,
+        info: &mut AfdPollInfo,
+        iosb: &mut IO_STATUS_BLOCK,
+        apccontext: PVOID,
+    ) -> io::Result<bool> {
+        unsafe {
+            let info_ptr: PVOID = transmute(info);
+            iosb.u.Status = STATUS_PENDING;
+            let status = NtDeviceIoControlFile(
+                self.fd.as_raw_handle(),
+                null_mut(),
+                None,
+                apccontext,
+                iosb,
+                IOCTL_AFD_POLL,
+                info_ptr,
+                size_of::<AfdPollInfo>() as u32,
+                info_ptr,
+                size_of::<AfdPollInfo>() as u32,
+            );
+            match status {
+                STATUS_SUCCESS => Ok(true),
+                STATUS_PENDING => Ok(false),
+                _ => Err(io::Error::from_raw_os_error(
+                    RtlNtStatusToDosError(status) as i32
+                )),
+            }
+        }
+    }
+
+    pub fn cancel(&self, iosb: &mut IO_STATUS_BLOCK) -> io::Result<()> {
+        unsafe {
+            if iosb.u.Status != STATUS_PENDING {
+                return Ok(());
+            }
+
+            let mut cancel_iosb: IO_STATUS_BLOCK = std::mem::zeroed();
+            let status = NtCancelIoFileEx(self.fd.as_raw_handle(), iosb, &mut cancel_iosb);
+            if status == STATUS_SUCCESS || status == STATUS_NOT_FOUND {
+                return Ok(());
+            }
+            Err(io::Error::from_raw_os_error(
+                RtlNtStatusToDosError(status) as i32
+            ))
+        }
+    }
+}
+
+pub const AFD_POLL_RECEIVE: u32 = 0x0001;
+pub const AFD_POLL_RECEIVE_EXPEDITED: u32 = 0x0002;
+pub const AFD_POLL_SEND: u32 = 0x0004;
+pub const AFD_POLL_DISCONNECT: u32 = 0x0008;
+pub const AFD_POLL_ABORT: u32 = 0x0010;
+pub const AFD_POLL_LOCAL_CLOSE: u32 = 0x0020;
+pub const AFD_POLL_ACCEPT: u32 = 0x0080;
+pub const AFD_POLL_CONNECT_FAIL: u32 = 0x0100;
+
+pub fn eventflags_to_afd_events(epoll_events: u32) -> u32 {
```

Comment:
I don't know if it makes sense for this code, but you don't have to convert IOCP early anymore, this can now be done in `Event`.

Reply:
Removed all epoll flags.